app/src/org/commcare/tasks/PrimeEntityCacheHelper.kt

```diff
@@ -65,12 +56,11 @@ class PrimeEntityCacheHelper private constructor() : Cancellable {
                     primeEntityCacheRequest
                 )
         }
+    }
 
-        @JvmStatic
-        fun cancelWork() {
-            instance?.cancel()
-            WorkManager.getInstance(CommCareApplication.instance()).cancelUniqueWork(PRIME_ENTITY_CACHE_REQUEST)
-        }
+    fun cancelWork() {
+        cancel()
+        WorkManager.getInstance(CommCareApplication.instance()).cancelUniqueWork(PRIME_ENTITY_CACHE_REQUEST)
```

Comment:
No need to address if I'm overthinking, but if the spin-down process for this getting executed takes some time, will it already be out of the queue reliably by the next worker? If not, should this be keyed against something like the app id / work type as well? It's probably a non-issue since an old worker will error out anyway when its database goes away.

Reply:
I would think so but probably better to not rely on assumptions and also it's more consistent with other app specific requests - https://github.com/dimagi/commcare-android/pull/2928/commits/99329d3292b941081cdead072fe559e9c193076e and https://github.com/dimagi/commcare-android/pull/2928/commits/89a054f50ac7a00baa2ed69d02d6a9592f1ef0e0